### PR TITLE
Update setup configuration and create initial tests

### PR DESCRIPTION
**Pull request by SmileyChris**
This whole project has no tests. Here are some base graphql_ws tests, and basic tests for gevent, aiohttp and django_channels.

Rather than use an all-in-one `requirements_dev.txt` file, I've switched to using the optional install functionality of setuptools. Developers should install the project like `pip install -e .[dev]`, maintainers (who also want to run tests) can do `pip install -e .[dev,maintainer]`.

### CODE REVIEW

File: graphql_ws/__init__.py [hunk 1]
```diff
@@ -7,6 +7,4 @@
 __version__ = '0.3.1'
 
 
-from .base import BaseConnectionContext, BaseSubscriptionServer
-
-__all__ = ['BaseConnectionContext', 'BaseSubscriptionServer']
```
Comment (KingDarBoja) [inline graphql_ws/__init__.py:12]: As far as I know, this line is supposed to export only the declared functions and "avoid" any other internal import from those modules.

File: setup.cfg [hunk 2]
```diff
@@ -1,23 +1,92 @@
...
+    Programming Language :: Python :: 2
+    Programming Language :: Python :: 2.6
+    Programming Language :: Python :: 2.7
+    Programming Language :: Python :: 3
+    Programming Language :: Python :: 3.3
+    Programming Language :: Python :: 3.4
+    Programming Language :: Python :: 3.5
+    Programming Language :: Python :: 3.6
+    Programming Language :: Python :: 3.7
+    Programming Language :: Python :: 3.8
```
Comment (KingDarBoja) [inline setup.cfg:26]: wasn't it better to keep this kind of information separate on the `setup.py` file and keep only formatting config on this file?

File: graphql_ws/__init__.py (hunk 1 above)
Comment (SmileyChris) [reply graphql_ws/__init__.py:12]: There's nothing that use these imports internally, the code all references the explicit `base`. These two classes aren't really usable externally, except perhaps as the base to your own third-party server. For a project that has languished so long, I don't see that being too much of a concern though - people can just update their code to import from the explicit base module too.

File: setup.cfg (hunk 2 above)
Comment (SmileyChris) [reply setup.cfg:26]: eh, I dunno it's all bikeshed painting. I like having nothing in my setup.py personally (eventually we won't need it, hopefully :D)

**Comment by SmileyChris** [comment]
Thanks for the review @KingDarBoja. I've got some more changes ready to deduplicate all the shared code and black the project, but wanted to do it in chunks rather than 1 giant PR

**Pull request merged**